packages/shared/src/like-anchor.ts

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2020 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+import { UpdatingElement, property, TemplateResult, html } from 'lit-element';
+import { ifDefined } from 'lit-html/directives/if-defined.js';
+
+type Constructor<T = object> = {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    new (...args: any[]): T;
+    prototype: T;
+};
+
+export interface LikeAnchorInterface {
+    download?: string;
+    label?: string;
+    href?: string;
+    target?: '_blank' | '_parent' | '_self' | '_top';
+    renderAnchor(options: {
+        id: string;
+        anchorContent?: TemplateResult | TemplateResult[];
+    }): TemplateResult;
+}
```

Comment:
Out of curiosity why is this interface needed? It feels like there's some intent here that would be good to add in a comment (or a common pattern being used I'm just ignorant on).

Reply:
Namely that it's needed in like 34 below... and so that the properties and methods therein can be seen by elements this is mixed into. Other than that it probably doesn't need to exported, but I was matching style to the `ObserveSlotText` mixin pattern. I don't have enough Typescript magic to say whether there is other intent that this could fill or not.

Reply:
Yep you have it down. Its just to enhance the type exported by the mixin so that the constructed object has these properties. Probably doesn't need export its true, but might be handy in some cases where we want to accept anything that is 'anchor like'?

Reply:
Ahhh -- thanks.